src/Microsoft.DotNet.Interactive.Http.Tests/ParserTests.NamedRequest.cs

```diff
@@ -0,0 +1,233 @@
+
+            result.SyntaxTree.RootNode.DescendantNodesAndTokens().OfType<HttpCommentNamedRequestNode>().Should().HaveCount(1);
+        }
+
+        [Fact]
+        public void named_request_can_reference_content_type_header()
+        {
+            var code = """
+                # @name login 
+
+                POST {{baseUrl}}/api/login HTTP/1.1 
+
+                Content-Type: application/x-www-form-urlencoded 
+
+
+
+                name=foo&password=bar 
+
+
+
+                ###
+                """;
+
+            var result = Parse(code);
+
+            var namedRequest = result.SyntaxTree.RootNode.DescendantNodesAndTokens().OfType<HttpCommentNamedRequestNode>().Single();
+
+            namedRequest.ValueNode.Text.Should().Be("login");
+        }
+
+        [Fact]
+        public void named_request_with_two_names_produces_an_error()
+        {
+            var code = """
+                # @name login two
+
+                POST {{baseUrl}}/api/login HTTP/1.1 
+
+                Content-Type: application/x-www-form-urlencoded 
+
+
+
+                name=foo&password=bar 
+
+
+
+                ###
+                """;
+
+            var result = Parse(code);
+
+            var namedRequest = result.SyntaxTree.RootNode.DescendantNodesAndTokens().OfType<HttpCommentNamedRequestNode>().Single();
+
+            namedRequest.GetDiagnostics().Should().HaveCount(1);   
+        }
+
+        [Fact]
+        public void named_request_with_special_characters_produces_an_error()
+        {
+            var code = """
+                # @name tes!
+
+                POST {{baseUrl}}/api/login HTTP/1.1 
+
+                Content-Type: application/x-www-form-urlencoded 
+
+
+
+                name=foo&password=bar 
+
+
+
+                ###
+                """;
+
+            var result = Parse(code);
+
+            var namedRequest = result.SyntaxTree.RootNode.DescendantNodesAndTokens().OfType<HttpCommentNamedRequestNode>().Single();
+
+            namedRequest.GetDiagnostics().Should().HaveCount(1);
+        }
+
+        [Theory]
+        [InlineData("login")]
+        [InlineData("   login   ")]
+        [InlineData("  login")]
+        [InlineData("login   ")]
+        [InlineData("login \r\n")]
+        public void named_request_with_various_spaces_are_parsed_correctly(string name)
+        {
+            var code = $$$"""
+                # @name {{{name}}}
+
+                POST {{baseUrl}}/api/login HTTP/1.1 
+
+                Content-Type: application/x-www-form-urlencoded 
+
+
+
+                name=foo&password=bar 
+
+
+
+                ###
+                """;
+
+            var result = Parse(code);
+
+            var namedRequest = result.SyntaxTree.RootNode.DescendantNodesAndTokens().OfType<HttpCommentNamedRequestNode>().Single();
+
+            namedRequest.ValueNode.Text.Should().Be("login");
+        }
+
+        [Fact]
+        public void named_request_with_slash_for_comment_start()
```

Comment:
Test names are more informative when they express the expectation. Maybe something like `named_request_comments_can_use_double_slash_prefix`.